util/lock.go

```diff
@@ -0,0 +1,173 @@
+package util
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"github.com/timescale/prometheus-postgresql-adapter/log"
+	"strconv"
+	"sync"
+	"time"
+)
+
+const (
+	connCheckInterval = time.Millisecond * 250
+)
+
+// PgAdvisoryLock is implementation of leader election protocol based on PostgreSQL advisory locks. All adapters withing a HA group are trying
```

Comment:
While not strictly wrong, I think calling this a "leader election protocol" is a bit of a stretch as I think most people would associate this with some kind of distributed algorithm (e.g., quorum-based).

Reply:
I have a feeling that locks are pretty common way to coordinate and implement leader election (Consul, Zookeeper), however I am open for better wording if you have some idea?

Reply:
Yes, but the point is that lock services like Zookeeper and Consul both implement a distributed leader election protocol (such as Raft or Paxos) among their respective groups/ensembles. That's what is typically meant by a "leader election protocol", specifically. I guess what trips me up a bit is the use of "protocol". Just removing "protocol" would lessen the confusion I think, e.g.,. `PgAdvisoryLock is implementation of leader election using PostgreSQL advisory locks`.